src/Raven.Server/Documents/Handlers/Admin/Processors/Configuration/AbstractAdminConfigurationHandlerProcessorForPutSettings.cs

```diff
@@ -46,4 +54,28 @@ public override async ValueTask ExecuteAsync()
 
         RequestHandler.NoContentStatus(HttpStatusCode.Created);
     }
+    
+    private List<string> GetUpdatedSettingsKeys(Dictionary<string, string> updatedSettings)
+    {
+        var keys = new List<string>();
+
+        using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+        using (context.OpenReadTransaction())
+        {
+            foreach (var kvp in updatedSettings)
+            {
+                var databaseRecord = ServerStore.Cluster.ReadDatabase(context, RequestHandler.DatabaseName, out var _);
```

Comment:
please note that you are reading database record for each key instead of once, also it would be better to read raw database record IMO?

Reply:
Changed